foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/databaseaccess/DatabaseAccessor.java

```diff
@@ -1383,6 +1383,9 @@ protected Object getObjectThroughOptimizedDataConversion(ResultSet resultSet, Da
         } else if ((fieldType == ClassConstants.SHORT) || (fieldType == ClassConstants.PSHORT)) {
             value = Short.valueOf(resultSet.getShort(columnNumber));
             isPrimitive = ((Short)value).shortValue() == 0;
+        } else if ((fieldType == ClassConstants.BOOLEAN) || (fieldType == ClassConstants.PBOOLEAN))  {
```

Comment:
I think the `allowResultTypeConversion()` property check needs to be included. This property is meant to protect customers who have written their applications to expect the Integer value, instead of the type Boolean. Without this change, production environments could see breaking, ClassCastException changes after updating to a newer version of 2.7.x. I think this is the safest approach to nonfailing, behavior changes that are not dictated by the spec.

Reply:
In latest commit I decided to remove `eclipselink.allow-result-type-conversion` property and related methods due a my and @lukasj fix https://github.com/eclipse-ee4j/eclipselink/pull/983 . As I checked sources, this property and `org.eclipse.persistence.sessions.Project#allowResultTypeConversion()` related method was used only in removed code.
As this property is not included in any release I think, that we can remove it.
If there is any reason why to keep it there let me know.

Reply:
#983 is MASTER, which I agree is safe to NOT NEED the property as there had not been a release yet. This is 2.7, which has many releases already. Your reasoning for removing the property is flawed

Reply:
Customers/users using 2.7.x could have written applications which are dependent on the `eclipselink.allow-result-type-conversion` property.  Updates to a released version should not break existing users.  Is there an exceptionally good reason for removing this property which justifies this risk to existing users?  Otherwise, it should remain.